engines/nancy/detection.cpp

```diff
@@ -284,44 +253,7 @@ static const Nancy::NancyGameDescription gameDescriptions[] = {
 	{ // MD5 by waltervn
 		{
 			"nancy5", nullptr,
-			{
-				{ "data1.hdr", 0, "261105fba2a1226eedb090c2ce79fd35", 284091 },
-				{ "data1.cab", 0, "7d27bb947ef7305831f1faaf1512a598", 1446301 },
-				{ "data2.cab", 0, "00719c86cab733c1094b27079ce030f3", 145857935 },
-				AD_LISTEND
-			},
-			Common::EN_ANY,
-			Common::kPlatformWindows,
-			ADGF_UNSTABLE | ADGF_DROPPLATFORM | Nancy::GF_COMPRESSED,
-			GUIO0()
-		},
-		Nancy::kGameTypeNancy5
-	},
-	{ // MD5 by clone2727
-		{
-			"nancy5", nullptr,
-			{
-				{ "data1.hdr", 0, "258e27792fa7cc7a7125fd74d89f8487", 284091 },
-				{ "data1.cab", 0, "70433b30b6114031d54d0c991ad44577", 1446301 },
```

Comment:
Hmmmm... It seems to me that you are killing these variants. I mean, these archives as a total do differ. Is that the case?

If this is the case, then I would suggest still keeping a distinct archive like `data1.cab` as a second file in the detection entry.

Reply:
Apart from program files, IS cabs contain a lot of extra data - installer strings in multiple languages, image banners for the installer, registry keys, optional libraries, etc etc. I do not have both of the variants to compare, but it's a pretty safe bet that there's no differences in the actual game data and the discrepancy is in the aforementioned extra data. HeR had a habit of re-releasing their games in various bundles, so having a slightly installer between versions, but identical game data is not out of the expected.
Also, since ciftree.dat (the file we use for detection) contains nearly all the game data except for music and some videos, we know for a fact there is no difference in game logic between those versions - if there were, the md5s would be different :)

Reply:
It is not about that. Once we have the integrity checking system in-place, it would require to have separate entries, so we could definitely tell that a game is complete and in distinct.

E.g. we would need to collect all variants in the detection tables.

Also, with this in light, I'd like to ask if possible, leave the current detection entries intact, because I am afraid it will be very difficult to port this archive support to Python. Or we decide to switch the scan utility to C++.

Reply:
Right, I forgot about the integrity system... In this case your proposal to keep a .cab file in the entry would work fine; the integrity system itself wouldn't need to worry about what's inside an archive, and could just skip files with an A: prefix, since any changes in the files within would be reflected in the checksum of the containing archive.

Only issue is that some of the entries in the nancy engine were made ~10 years ago by other contributors, and I'd need their help with extracting the checksums of the ciftree.dat files the engine actually cares about; I'm assuming they'd be the same, but I can't know since I don't own those variants myself. I think for now commenting out the entries I can't upgrade to the new system and then re-activating them with the new data would be the best course of action; I'd like to not have any cabfile-only entries active, since the whole point of this PR is to get rid of the false positives they cause.